(DOC +) Add resolutions to Allocation Explain examples (#108263)
elastic/elasticsearch: docs/reference/cluster/allocation-explain.asciidoc

<3> Whether to allocate the shard.
<4> Whether to allocate the shard to the particular node.
<5> The decider which led to the `no` decision for the node.

> <6> An explanation as to why the decider returned a `no` decision, with a helpful hint pointing to the setting that led to the decision. For the example above, a newly created index has <<indices-get-settings,an index setting>> requiring it to only be allocated to a node named `nonexistent_node`, which does not exist so the index is unable to allocate.

Reviewer: because this is a footnote, we don't have to say "above"

```suggestion
<6> An explanation as to why the decider returned a `no` decision, with a helpful hint pointing to the setting that led to the decision. In this example, a newly created index has <<indices-get-settings,an index setting>> that requires that it only be allocated to a node named `nonexistent_node`, which does not exist, so the index is unable to allocate.
```

----
// NOTCONSOLE

> TIP: If a shard is unassigned with an allocation status of `no_valid_shard_copy`, then you should <<fix-cluster-status-recover-nodes,make sure that all nodes are in the cluster>>. If the node containing a primary shard is lost, then you can <<fix-cluster-status-restore,recover the data for the shard>>.

Reviewer 2: Extreme pedantry warning: if you lose the node containing a primary shard then normally Elasticsearch will just promote one of the remaining replicas to become the primary without any fuss. We really should talk about losing the in-sync copies here.

```suggestion
TIP: If a shard is unassigned with an allocation status of `no_valid_shard_copy`, then you should <<fix-cluster-status-recover-nodes,make sure that all nodes are in the cluster>>. If all the nodes containing in-sync copies of a shard are lost, then you can <<fix-cluster-status-restore,recover the data for the shard>>.
```

> The shard in the example above errors `no_valid_shard_copy` due to `NODE_LEFT`. The first step to recover is to make sure all nodes are in the cluster. Then, if the error continues after a <<cluster-reroute,cluster reroute>>, the data will need to be <<snapshots-restore-snapshot,restored from snapshot>>.

Reviewer: I think this info is valuable, but I don't think that it's the appropriate location for the recovery steps. We have a [page that talks about recovering lost nodes](https://www.elastic.co/guide/en/elasticsearch/reference/current/red-yellow-cluster-status.html#fix-cluster-status-recover-nodes) - we might consider linking to that doc, and enhancing it as needed. This way, all of the info is centralized. Let me know what you think.

Reviewer 3: 👋🏽 I'm on board to either page as long as we can write an actionable commentary for this response example. (I helped Annie write her first doc PR!) The Support back story is that users search for the explicit error and `no_valid_shard_copy` only shows on this page. 

We could just link or also add the error's literal string on the [recovery data for a lost primary shard](https://www.elastic.co/guide/en/elasticsearch/reference/current/red-yellow-cluster-status.html#fix-cluster-status-restore). If we cross-link/pollinate, could y'all also kindly consider reformatting the destination so these don't look like sequential steps (which'd destroy your data right after recovering it?):
<img width="445" alt="image" src="https://github.com/elastic/elasticsearch/assets/26751266/f50bec37-f826-47ac-82a5-67c42ab8f84a">

> The shard in the example above errors `no_valid_shard_copy` due to `NODE_LEFT`. The first step to recover is to make sure all nodes are in the cluster. Then, if the error continues after a <<cluster-reroute,cluster reroute>>, the data will need to be <<snapshots-restore-snapshot,restored from snapshot>>.

Reviewer 2: The `NODE_LEFT` bit is a distraction - it describes why the shard became unassigned in the first place, but that's not normally relevant to the process of working out how to recover. I'd suggest not mentioning it.

We definitely shouldn't suggest performing a cluster reroute either. If that's needed, it'll say so explicitly in the API response. It doesn't say so in the example, so it won't be needed. We might at some point actually reject such unnecessary calls, see https://github.com/elastic/elasticsearch/issues/81231 for more information.

More generally, I think the best place for this sort of advice is the [relevant troubleshooting page](https://www.elastic.co/guide/en/elasticsearch/reference/current/red-yellow-cluster-status.html). That covers a lot more cases than just this one example. Can we just link to that page instead?

Reviewer 3: ACK on just linking the other page; sounds like we're discussing in thread above, cheers. 

I will highlight for the example, `NODE_LEFT` is applicable for the caveat "make sure all nodes are in the cluster" because that's the first resolution step and a common user gotcha IMO. 

However, agreed we shouldn't necessarily need a Cluster Reroute, Support's just used to recommending that to get the `retry_failed=true` option for the cluster to reconsider faster. (Or it does nothing but it seems to do something 🙂 .)

Reviewer: perhaps this could be a tip with links, like the following:

```suggestion
TIP: If a shard is unassigned with an allocation status of `no_valid_shard_copy`, then you should <<fix-cluster-status-recover-nodes,make sure that all nodes are in the cluster>>. If the node containing a primary shard is lost, then you can <<fix-cluster-status-restore,recover the data for the shard>>.
```
